sim/battle-actions.ts

```diff
@@ -1604,6 +1604,7 @@ export class BattleActions {
 			defBoosts = 0;
 		}
 
+		if (move.isFutureMove && !source.isActive && move.recentForme) attacker.setSpecies(move.recentForme);
```

Comment:
Doesn't this mean that a Ditto that transformed into Jirachi to use Doom Desire and then was switched out, after hitting the Ditto will go back to the field as a Jirachi?

Reply:
It appears that this isn't the case when I run a server using this code locally. I'm not entirely certain why this is, I would guess that there are certain mechanisms for changing a Pokemon back to its base forme upon switching in.